library/src/device/generator/generator.kernel.hpp

```diff
@@ -342,21 +347,25 @@ namespace StockhamGenerator
             /*=======================================================*/
             std::string loop;
             loop += "\tif(dim == 1){\n";
-            loop += "\t\t" + offset_name + " += counter_mod*" + stride_name + "[1];\n";
+            loop += "\t\t" + offset_name1 + " += counter_mod*" + stride_name1 + "[1];\n";
+            if (output == true) loop += "\t\t" + offset_name2 + " += counter_mod*" + stride_name2 + "[1];\n";
```

Comment:
@b-sumner I don't see the problem, there's the offset_name2 string before the '+=' , which would look like 'oOffset += ' in the generated line.
@tingxingdong you tested this code right?